src/main/java/appeng/tile/powersink/AEBasePoweredTile.java

```diff
@@ -19,7 +19,308 @@
+			{
+				amt = this.getInternalCurrentPower() - this.getInternalMaxPower();
+				this.setInternalCurrentPower( this.getInternalMaxPower() );
+				return amt;
+			}
+
+			return 0;
+		}
+	}
+
+	protected void PowerEvent( final PowerEventType x )
+	{
+		// nothing.
+	}
+
+	@Override
+	public final double getAEMaxPower()
+	{
+		return this.getInternalMaxPower();
+	}
+
+	@Override
+	public final double getAECurrentPower()
+	{
+		return this.getInternalCurrentPower();
+	}
+
+	@Override
+	public final boolean isAEPublicPowerStorage()
+	{
+		return this.isInternalPublicPowerStorage();
+	}
+
+	@Override
+	public final AccessRestriction getPowerFlow()
+	{
+		return this.getInternalPowerFlow();
+	}
+
+	@Override
+	public final double extractAEPower( final double amt, final Actionable mode, final PowerMultiplier multiplier )
+	{
+		return multiplier.divide( this.extractAEPower( multiplier.multiply( amt ), mode ) );
+	}
+
+	protected double extractAEPower( double amt, final Actionable mode )
+	{
+		if( mode == Actionable.SIMULATE )
+		{
+			if( this.getInternalCurrentPower() > amt )
+			{
+				return amt;
+			}
+			return this.getInternalCurrentPower();
+		}
+
+		final boolean wasFull = this.getInternalCurrentPower() >= this.getInternalMaxPower() - 0.001;
+		if( wasFull && amt > 0.001 )
+		{
+			this.PowerEvent( PowerEventType.REQUEST_POWER );
+		}
+
+		if( this.getInternalCurrentPower() > amt )
+		{
+			this.setInternalCurrentPower( this.getInternalCurrentPower() - amt );
+			return amt;
+		}
+
+		amt = this.getInternalCurrentPower();
+		this.setInternalCurrentPower( 0 );
+		return amt;
+	}
+
+	public double getInternalCurrentPower()
+	{
+		return this.internalCurrentPower;
+	}
+
+	public void setInternalCurrentPower( final double internalCurrentPower )
+	{
+		this.internalCurrentPower = internalCurrentPower;
+	}
+
+	public double getInternalMaxPower()
+	{
+		return this.internalMaxPower;
+	}
+
+	public void setInternalMaxPower( final double internalMaxPower )
+	{
+		this.internalMaxPower = internalMaxPower;
+	}
+
+	private boolean isInternalPublicPowerStorage()
+	{
+		return this.internalPublicPowerStorage;
+	}
+
+	public void setInternalPublicPowerStorage( final boolean internalPublicPowerStorage )
+	{
+		this.internalPublicPowerStorage = internalPublicPowerStorage;
+	}
+
+	private AccessRestriction getInternalPowerFlow()
+	{
+		return this.internalPowerFlow;
+	}
+
+	public void setInternalPowerFlow( final AccessRestriction internalPowerFlow )
+	{
+		this.internalPowerFlow = internalPowerFlow;
+	}
+
+	@Override
+	public void onReady()
+	{
+		super.onReady();
+
+		this.ic2Sink.onLoad();
+	}
+
+	@Override
+	public void onChunkUnload()
+	{
+		super.onChunkUnload();
+
+		this.ic2Sink.onChunkUnload();
+	}
+
+	@Override
+	public void invalidate()
+	{
+		super.invalidate();
+
+		this.ic2Sink.invalidate();
+	}
+
+	@Override
+	public boolean hasCapability( Capability<?> capability, EnumFacing facing )
+	{
+		if( capability == Capabilities.FORGE_ENERGY )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this conditional structure or edit its code blocks so that they're not all the same. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS3923)